stratego.build.spoofax2/src/main/java/mb/stratego/build/spoofax2/Spoofax2ParseStratego.java

```diff
@@ -0,0 +1,113 @@
+import java.nio.charset.Charset;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.vfs2.FileObject;
+import org.metaborg.core.config.JSGLRVersion;
+import org.metaborg.core.language.ILanguage;
+import org.metaborg.core.language.ILanguageIdentifierService;
+import org.metaborg.core.language.ILanguageImpl;
+import org.metaborg.core.language.ILanguageService;
+import org.metaborg.core.language.IdentifiedResource;
+import org.metaborg.core.resource.IResourceService;
+import org.metaborg.spoofax.core.SpoofaxConstants;
+import org.metaborg.spoofax.core.syntax.ISpoofaxSyntaxService;
+import org.metaborg.spoofax.core.unit.ISpoofaxInputUnit;
+import org.metaborg.spoofax.core.unit.ISpoofaxParseUnit;
+import org.metaborg.spoofax.core.unit.ISpoofaxUnitService;
+import org.spoofax.interpreter.terms.IStrategoAppl;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.spoofax.interpreter.terms.ITermFactory;
+import org.spoofax.terms.io.binary.TermReader;
+import org.spoofax.terms.util.TermUtils;
+
+import mb.pie.api.ExecException;
+import mb.stratego.build.strincr.ParseStratego;
+import mb.stratego.build.termvisitors.DisambiguateAsAnno;
+import mb.stratego.build.util.StrIncrContext;
+
+public class Spoofax2ParseStratego implements ParseStratego {
+    private final IResourceService resourceService;
+    private final ILanguageIdentifierService languageIdentifierService;
+    private final ILanguageService languageService;
+    private final ITermFactory termFactory;
+    private final ISpoofaxUnitService unitService;
+    private final ISpoofaxSyntaxService syntaxService;
+    private final StrIncrContext strContext;
+
+    @Inject public Spoofax2ParseStratego(IResourceService resourceService,
+        ILanguageIdentifierService languageIdentifierService, ILanguageService languageService,
+        ITermFactory termFactory, ISpoofaxUnitService unitService, ISpoofaxSyntaxService syntaxService,
+        StrIncrContext strContext) {
+        this.resourceService = resourceService;
+        this.languageIdentifierService = languageIdentifierService;
+        this.languageService = languageService;
+        this.termFactory = termFactory;
+        this.unitService = unitService;
+        this.syntaxService = syntaxService;
+        this.strContext = strContext;
+    }
+
+    @Override public IStrategoTerm parse(InputStream inputStream, Charset charset, @Nullable String path)
+        throws Exception {
+        final @Nullable FileObject inputFile;
+        if(path != null) {
+            inputFile = resourceService.resolve(path);
+        } else {
+            inputFile = null;
+        }
+
+        @Nullable ILanguageImpl strategoLangImpl = null;
+        @Nullable ILanguageImpl strategoDialect = null;
+        if(inputFile != null) {
+            final @Nullable IdentifiedResource identified = languageIdentifierService.identifyToResource(inputFile);
```

Comment:
This looks a lot simpler than what I had. Do you think this will work the same as my cobbled-together solution?

Reply:
Yes it should work the same